src/Fade.js

```diff
@@ -1,128 +1,62 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import Transition from 'react-transition-group/Transition';
 import { mapToCssModules, omit } from './utils';
 
 const propTypes = {
+  ...Transition.propTypes,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node
+  ]),
+  tag: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
   baseClass: PropTypes.string,
-  baseClassIn: PropTypes.string,
-  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+  baseClassActive: PropTypes.string,
   className: PropTypes.string,
   cssModule: PropTypes.object,
-  transitionAppearTimeout: PropTypes.number,
-  transitionEnterTimeout: PropTypes.number,
-  transitionLeaveTimeout: PropTypes.number,
-  transitionAppear: PropTypes.bool,
-  transitionEnter: PropTypes.bool,
-  transitionLeave: PropTypes.bool,
-  onLeave: PropTypes.func,
-  onEnter: PropTypes.func,
 };
 
 const defaultProps = {
   tag: 'div',
   baseClass: 'fade',
-  baseClassIn: 'show',
-  transitionAppearTimeout: 0,
-  transitionEnterTimeout: 0,
-  transitionLeaveTimeout: 0,
-  transitionAppear: true,
-  transitionEnter: true,
-  transitionLeave: true
+  baseClassActive: 'show',
+  timeout: 150,
```

Comment:
It appears bootstrap default is 300ms

https://github.com/twbs/bootstrap/blob/v4-dev/scss/_variables.scss#L688

We should definitely comment this (briefly here and more thoroughly in the example/documentation) as the timeout here also requires updating `$modal-transition` in custom bootstrap builds as this is important for syncing the CSS UI with the timeouts here.


EDIT: I apologize. I referenced the Modal timeouts against the Fade here.  Bootstrap's Fade components are all 150 so this is correct. The modal is separate which uses 300ms/150ms. Could we at least add documentation in both these locations in source and in documentation examples as its real important that these sync otherwise there will be huge jumps where the transition end timeouts prior or the CSS transition is still continuing.

* Modal timeout changes need to update: `$modal-transition`
* Fade component changes (e.g. Alert) need to update: `$transition-fade`